build.py

```diff
@@ -185,10 +185,21 @@
             "test_requirements.txt",
         ]
         subprocess.run(pip_install_args, check=True, text=True, cwd=pip_src)
-        pip_install_args = [python_bin, "-m", "pip", "install", "-e", "."]
+        pip_install_args = [
+            python_bin,
+            "-m",
+            "pip",
+            "install",
+            "--force-reinstall",
+            "--no-index",
+            "--find-links=" + wheels_dir,
+            "qsharp",
```

Comment:
For me the `python build.py --pip` fails without the following:
```suggestion
            "qsharp_python",
```

Reply:
That's odd. I wonder if the package name is getting picked up from Cargo.toml (instead of pyproject.toml) somehow. I'll take another look.

Reply:
Oh, that was my local change. I forgot to mention that I rebased my local changes on top of yours. And in my changes I made the name identical in Cargo.toml and pyproject.toml. 
My bad, disregard.

Reply:
Thanks for clarifying. The [documentation for the build backend](https://github.com/PyO3/maturin#python-metadata) (Maturin) says that it merges metadata from Cargo.toml and pyproject.toml, with pyproject.toml taking precedence. So, for the purposes of the Python package name, the name in Cargo.toml should be getting ignored.

I believe the reason I ended up changing the name in Cargo.toml to `qsharp_python` is that this package is also referenced from the Rust workspace, and we didn't want a generic name like `qsharp` in that context. ([see original PR](https://github.com/microsoft/qsharp/pull/183/files/ea8def5e1c3a5faf4a554bc47e3625bebe27c65d#r1178521419)). In practice, I don't think the name in Cargo.toml is going to matter too much since we're never going to publish this Rust crate by itself.